### PR TITLE
Remove embedded to oneToOne, add new Migrations

### PR DESCRIPTION
**Pull request by Tomanhez**

### CODE REVIEW

File: src/Entity/CreditMemoInterface.php [hunk 1]
```diff
@@ -18,15 +18,15 @@ public function getCurrencyCode(): string;
 
     public function getLocaleCode(): string;
 
-    public function getChannel(): CreditMemoChannel;
+    public function getChannel(): CreditMemoChannelInterface;
```
Comment (lchrusciel) [inline src/Entity/CreditMemoInterface.php]:
```suggestion
    public function getChannel(): ChannelInterface;
```

File: src/Resources/config/doctrine/CreditMemo.orm.xml [hunk 2]
```diff
@@ -19,9 +19,29 @@
         <field name="comment" type="text" />
         <field name="issuedAt" column="issued_at" type="datetime" nullable="true" />
 
-        <embedded name="channel" class="Sylius\RefundPlugin\Entity\CreditMemoChannel" />
-        <embedded name="from" class="Sylius\RefundPlugin\Entity\CustomerBillingData" />
-        <embedded name="to" class="Sylius\RefundPlugin\Entity\ShopBillingData" />
+        <one-to-one field="channel" target-entity="Sylius\RefundPlugin\Entity\CreditMemoChannel">
+            <cascade>
+                <cascade-persist/>
```
Comment (lchrusciel) [inline src/Resources/config/doctrine/CreditMemo.orm.xml]:
```suggestion
                <cascade-all/>
```

Comment (lchrusciel) [reply src/Resources/config/doctrine/CreditMemo.orm.xml]: Same for the rest of `cascade`

**Comment by AdamKasp** [comment]
One thing, you deleted `CreditMemoChannel` so IMO you should add to documentations or to README  some information about how to migrate from logic with `CreditMemoChannel` to logic with `ChannelInterface` . Thanks to that we will avoid something like "BC break"

File: spec/Entity/CreditMemoSpec.php [hunk 3]
```diff
@@ -24,7 +24,7 @@ function let(): void
             1000,
             'USD',
             'en_US',
-            new CreditMemoChannel('WEB-US', 'United States', 'Linen'),
+            new Channel(),
```
Comment (GSadee) [inline spec/Entity/CreditMemoSpec.php]:
```suggestion
            $channel,
```
and put `ChannelInterface $channel` in `let()` method

File: spec/Entity/CreditMemoSpec.php [hunk 4]
```diff
@@ -70,7 +70,7 @@ function it_has_locale_code(): void
 
     function it_has_channel(): void
     {
-        $this->getChannel()->shouldBeLike(new CreditMemoChannel('WEB-US', 'United States', 'Linen'));
+        $this->getChannel()->shouldHaveType(Channel::class);
```
Comment (GSadee) [inline spec/Entity/CreditMemoSpec.php]:
```suggestion
        $this->getChannel()->shouldHaveType(ChannelInterface::class);
```

File: src/Resources/config/doctrine/CustomerBillingData.orm.xml [hunk 5]
```diff
@@ -6,7 +6,12 @@
         xsi:schemaLocation="http://doctrine-project.org/schemas/orm/doctrine-mapping
                             http://doctrine-project.org/schemas/orm/doctrine-mapping.xsd"
 >
-    <embeddable name="Sylius\RefundPlugin\Entity\CustomerBillingData">
+
```
Comment (GSadee) [inline src/Resources/config/doctrine/CustomerBillingData.orm.xml]: Unnecessary blank line

File: spec/Entity/CreditMemoSpec.php [hunk 6]
```diff
@@ -70,7 +70,7 @@ function it_has_locale_code(): void
 
     function it_has_channel(): void
```
Comment (lchrusciel) [inline spec/Entity/CreditMemoSpec.php]:
```suggestion
    function it_has_channel(ChannelInterface $channel): void
```

File: spec/Entity/CreditMemoSpec.php [hunk 7]
```diff
@@ -70,7 +70,7 @@ function it_has_locale_code(): void
...
     {
-        $this->getChannel()->shouldBeLike(new CreditMemoChannel('WEB-US', 'United States', 'Linen'));
+        $this->getChannel()->shouldHaveType(ChannelInterface::class);
```
Comment (lchrusciel) [inline spec/Entity/CreditMemoSpec.php]:
```suggestion
        $this->getChannel()->shouldReturn($channel);
```

File: docs/improvements.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+### Improvements
```
Comment (GSadee) [inline docs/improvements.md]: There is no reference to this file anywhere

File: docs/improvements.md [hunk 9]
```diff
@@ -0,0 +1,5 @@
...
+
+1. Removed ``CreditMemoChannel`` and replaced it to ``Sylius\Component\Core\Model\ChannelInterface``
```
Comment (GSadee) [inline docs/improvements.md]:
```suggestion
1. Removed ``CreditMemoChannel`` and replaced by ``Sylius\Component\Core\Model\ChannelInterface``.
```

File: docs/improvements.md [hunk 10]
```diff
@@ -0,0 +1,5 @@
...
+
+2. Replaced  ``CustomerBillingData``, ``ShopBillingData`` embedded to association and added interfaces ``CustomerBillingDataInterface``, ``ShopBillingDataInterface``
```
Comment (GSadee) [inline docs/improvements.md]: Sth like:
```suggestion
2. Replaced  ``CustomerBillingData`` and ``ShopBillingData`` value objects by entities with ``CustomerBillingDataInterface`` and ``ShopBillingDataInterface`` interfaces.
```

File: README.md [hunk 11]
```diff
@@ -72,6 +72,10 @@ This installation instruction assumes that you're using Symfony Flex. If you don
 [legacy installation instruction](docs/legacy_installation.md). However, we strongly encourage you to use
 Symfony Flex, it's much quicker! :)
 
+#### Improvements and changes!
+
+All improvements and changes you will find in [improvements](docs/improvements.md)
```
Comment (Zales0123) [inline README.md]: Maybe we should just create an UPGRADE file for `0.8.0` -> `0.9.0`?

File: src/Entity/CustomerBillingDataInterface.php [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\RefundPlugin\Entity;
+
+interface CustomerBillingDataInterface
+{
+    public function getId(): int;
```
Comment (Zales0123) [inline src/Entity/CustomerBillingDataInterface.php]: If we would like to be consistent with other getters, it should be `public function id(): int`

File: src/Entity/ShopBillingDataInterface.php [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\RefundPlugin\Entity;
+
+interface ShopBillingDataInterface
+{
+    public function getId(): int;
```
Comment (Zales0123) [inline src/Entity/ShopBillingDataInterface.php]: Same here

File: UPGRADE.md [hunk 14]
```diff
@@ -0,0 +1,7 @@
+### UPGRADE FROM 0.8.0 TO 0.9.0
+
+1. Removed ``CreditMemoChannel`` and replaced by ``Sylius\Component\Core\Model\ChannelInterface``.
+
+2. Replaced  ``CustomerBillingData`` and ``ShopBillingData`` value objects by entities with ``CustomerBillingDataInterface`` and ``ShopBillingDataInterface`` interfaces.
+
+3. Created `Version20190928200659.php`
```
Comment (GSadee) [inline UPGRADE.md]: Not necessary I guess

File: migrations/Version20190928200659.php [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+        $this->addSql('
+            UPDATE sylius_refund_credit_memo
+            INNER JOIN sylius_refund_shop_billing_data
+            SET sylius_refund_credit_memo.to_id = sylius_refund_shop_billing_data.id
+            WHERE sylius_refund_credit_memo.id = sylius_refund_shop_billing_data.id_credit_memo
+        ');
+        $this->addSql('
+            ALTER TABLE sylius_refund_shop_billing_data
+            DROP COLUMN id_credit_memo
+        ');
```
Comment (GSadee) [inline migrations/Version20190928200659.php]: Could be in one line

File: migrations/Version20190928200659.php [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+
+        $this->addSql('CREATE TABLE sylius_refund_embeddable_backup (id INT AUTO_INCREMENT NOT NULL, id_credit_memo VARCHAR(255) NOT NULL, channel_code VARCHAR(255) NOT NULL, from_customerName VARCHAR(255) NOT NULL, from_street VARCHAR(255) NOT NULL, from_postcode VARCHAR(255) NOT NULL, from_countryCode VARCHAR(255) NOT NULL, from_city VARCHAR(255) NOT NULL, from_company VARCHAR(255) DEFAULT NULL, from_provinceName VARCHAR(255) DEFAULT NULL, from_provinceCode VARCHAR(255) DEFAULT NULL, to_company VARCHAR(255) DEFAULT NULL, to_taxId VARCHAR(255) DEFAULT NULL, to_countryCode VARCHAR(255) DEFAULT NULL, to_street VARCHAR(255) DEFAULT NULL, to_city VARCHAR(255) DEFAULT NULL, to_postcode VARCHAR(255) DEFAULT NULL, PRIMARY KEY(id)) DEFAULT CHARACTER SET UTF8 COLLATE UTF8_unicode_ci ENGINE = InnoDB');
```
Comment (GSadee) [inline migrations/Version20190928200659.php]:
```suggestion
        $this->addSql('CREATE TABLE sylius_refund_embeddable_backup (id INT AUTO_INCREMENT NOT NULL, credit_memo_id VARCHAR(255) NOT NULL, channel_code VARCHAR(255) NOT NULL, from_customer_name VARCHAR(255) NOT NULL, from_street VARCHAR(255) NOT NULL, from_postcode VARCHAR(255) NOT NULL, from_country_code VARCHAR(255) NOT NULL, from_city VARCHAR(255) NOT NULL, from_company VARCHAR(255) DEFAULT NULL, from_province_name VARCHAR(255) DEFAULT NULL, from_province_code VARCHAR(255) DEFAULT NULL, to_company VARCHAR(255) DEFAULT NULL, to_tax_id VARCHAR(255) DEFAULT NULL, to_country_code VARCHAR(255) DEFAULT NULL, to_street VARCHAR(255) DEFAULT NULL, to_city VARCHAR(255) DEFAULT NULL, to_postcode VARCHAR(255) DEFAULT NULL, PRIMARY KEY(id)) DEFAULT CHARACTER SET UTF8 COLLATE UTF8_unicode_ci ENGINE = InnoDB');
```

**Comment by pamil** [comment]
Thanks, Tomasz! :tada:

File: src/Entity/ShopBillingData.php [hunk 17]
```diff
@@ -40,6 +44,11 @@ public function __construct(
         $this->postcode = $postcode;
     }
 
+    public function id(): int
```
Comment (lchrusciel) [inline src/Entity/ShopBillingData.php]: I'm not sure, if we want to force anyone to use int as an id

**Pull request merged**